### PR TITLE
[Bot Service] BREAKING CHANGE: `az bot create` change: Remove `--kind`, `--password`, `--lang` arguments. Add `--app-type`, `--tenant-id`, `--msi-resource-id` arguments

### PR DESCRIPTION
**Pull request by chinazhangchao**
**Related command**
az bot create

**Description**
The function of `az bot create` command is not corresponding with the azure devportal UI. So we plan to change the command parameters the same as devportal. This pr is to change the parameters same as 'Use existing app registration' type on devportal.

**Testing Guide**

Newly added test cases:

- test_botservice_registration_bot_create_singletenant_type
- test_botservice_registration_bot_create_multitenant_type
- test_botservice_registration_bot_create_userassignedmsi_type

Manually validate existing functionality

- run az bot create -g {} -n {} --appid {} --app-type SingleTenant --tenant-id {} -l {}
- run az bot create -g {} -n {} --appid {} --app-type MultiTenant -l {}
- run az bot create -g {} -n {} --appid {} --app-type UserAssignedMSI --tenant-id {} --msi-resource-id {} -l {}

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Bot Service

File: src/azure-cli/azure/cli/command_modules/botservice/_exception_handler.py [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
     from msrest.exceptions import ClientRequestError  # pylint: disable=import-error
-    if isinstance(ex, ErrorException):
+    if isinstance(ex, HttpResponseError):
         message = 'An error occurred. {0}: {1}'.format(
-            ex.error.error.code,
-            ex.error.error.message
+            ex.error.code,
+            ex.error.message
         )
         raise CLIError(message)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/botservice/_exception_handler.py:18]: CLIError is dropped. Please use error types defined in this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#available-error-types)

Comment (chinazhangchao) [reply src/azure-cli/azure/cli/command_modules/botservice/_exception_handler.py]: fixed

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**